web/pgadmin/tools/sqleditor/static/js/components/sections/ResultSet.jsx

```diff
@@ -380,12 +381,15 @@ export class ResultSetUtils {
   }
 
   saveData(reqData) {
-    return this.api.post(
+    const response = this.api.post(
       url_for('sqleditor.save', {
         'trans_id': this.transId
       }),
       JSON.stringify(reqData)
     );
+    // Set the commit flag to true
+    this.hasQueryCommitted = true;
```

Comment:
What if data has not been saved in case of any backend error? So, in case of an error too, this logic will fetch the data from the PG server while downloading the data